File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    let counts = [];
    const answer = [];
    for(let i = 0 ; i < progresses.length ; i++){
        let count = 0;
        while(progresses[i] < 100){
            progresses[i] += speeds[i];
            count++;
        }
        counts.push(count);
    }
    
    while(true){
        let front = counts[0];
        let deploy = 0;
        while(counts[0] <= front){
            deploy++;
            counts.shift();
        }
        answer.push(deploy);
        if(counts.length === 0) break;
    }
    return answer;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    dep_checker = [False] * len(progresses)

    while sum(answer) < len(progresses):
        # 하루 진행
        for i in range(len(progresses)):
            progresses[i] += speeds[i]
        
        # 마감 (waiting list check)
        if progresses[0] >= 100:
            dep_cnt = 0
            for j in range(len(progresses)):
                if progresses[j] < 100:
                    break
                else:
                    if dep_checker[j]:
                        pass
                    else:
                        dep_cnt += 1
                        dep_checker[j] = True
            if dep_cnt != 0:
            	answer.append(dep_cnt)
    
    return answer